transport/http/outbound.go

```diff
@@ -256,6 +256,12 @@ func (o *Outbound) call(ctx context.Context, treq *transport.Request) (*transpor
 
 	span.SetTag("http.status_code", response.StatusCode)
 
+	// Service name match validation, return yarpcerrors.CodeInternal error if not match
+	if match, resSvcName := checkServiceMatch(treq.Service, &response.Header); !match {
+		return nil, transport.UpdateSpanWithErr(span,
+			yarpcerrors.InternalErrorf("service name does not match, expected: %s, got: %s", treq.Service, resSvcName))
```

Comment:
nit: use `%q` here instead of `%s`. That will add double quotes around the name and escape it if needed. An advantage of that is that if either value is empty, it'll say,

    expected "", got "whatever"

Also, let's mention where these names are coming from in the error message:

    service name sent from the request does not match the service name received in the response: sent %q, got %q

Reply:
Resolved. Thanks for the suggestion.